server/py/framework/db/sqldb/db.py

```diff
@@ -5927,7 +5937,30 @@ def store_alert_activation(
             data=extra_data,
         )
 
-        self._upsert(session, [alert_activation_record])
+        # if reset_policy is MANUAL, we need to keep id to be able to update number_of_events
+        # when policy is reset
+        if alert_data.reset_policy == mlrun.common.schemas.alert.ResetPolicy.MANUAL:
+            return self._upsert_object_and_flush_to_get_field(
+                session, alert_activation_record, "id"
+            )
+        else:
+            return self._upsert(session, [alert_activation_record])
+
+    def update_alert_activation(
```

Comment:
```suggestion
    def update_alert_activation_number_of_events(
```

This is the only thing we allow updating